packages/flutter_tools/lib/src/cache.dart

```diff
@@ -41,7 +41,7 @@ class DevelopmentArtifact {
   static const DevelopmentArtifact iOS = DevelopmentArtifact._('ios');
 
   /// Artifacts required for web development.
-  static const DevelopmentArtifact web = DevelopmentArtifact._('web', unstable: true);
+  static const DevelopmentArtifact web = DevelopmentArtifact._('web');
```

Comment:
if stable/unstable is too coarse-grained, then maybe we could replace the `stable` field of `DevelopmentArtifact` with something like `final List<String> availableOn` which lists the branches that can use it.

Reply:
I think these should be tied to features in some way. Otherwise we risk the artifacts and features going out of sync again. Let me give something like that a try

Reply:
I added an option for tying an artifact to a Feature. This leaves the `unstable` flag since I didn't create a feature for each artifact class.